lib/qesapdeployment.pm

```diff
@@ -276,6 +279,8 @@ sub qesap_execute {
     $exec_log .= "_$args{cmd_options}" if ($args{cmd_options});
     $exec_log .= '.log.txt';
     $exec_log =~ s/[-\s]+/_/g;
+    # activate virtual environment
+    script_run("source " . QESAPDEPLOY_VENV . "/bin/activate");
```

Comment:
Correct. This was intended, covering for a case where `execute` is called in another context, without `configure` running the pip install with the venv creation before it (I.e. in another usecase). I was not sure that we should  abort if the venv is not there. Should I `assert_script_run` it after all?

Reply:
I am afraid I do not know enough about python virtual environments to properly reply. I simply thought we needed to be within the VENV before calling the `qesap.py` command on L285. What would happen if `source QESAPDEPLOY_VENV/bin/activate` fails and the command in L285 is invoked? Is it expected to fail as well?

Reply:
That depends on whether the base python installation has the required modules. If it does not, it will fail yes, but if it does it won't (i.e. if we use a different `configure` module for a different scenario, where all pip installations are run on the base installation, or use an image with everything pre-installed).

In the end, all that matters is: are we certain that every time the `qesap_execute` function runs, `qesap_pip_install` will have ran before it? If yes, I should make the activation mandatory, if not, it's better for it to stay as it is.